<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="rviz_file" default="nav_LP"/>
  <arg name="model" default="model"/>
  <arg name="update_rate" />
  

  <!-- <rosparam command="load" file="$(find arena_bringup)/params/flatland_deploy.yaml" /> -->

  <param name="step_size" value="0.1" />
  <param name="update_rate" value="$(arg update_rate)" />
  <param name="viz_pub_rate" value="5000" />

  <arg name="visualization" default="rviz" doc="[rviz, flatland]" />

  <remap from="/move_base_simple/goal" to="task_generator/set_goal" />

  <!-- launch flatland server -->
  <node name="flatland_server" pkg="flatland_server" type="flatland_server">

  </node>

  <!--  _____________ Robot Model _____________  -->
  <arg name="initial_pose_x" default="0.0"/>
  <arg name="initial_pose_y" default="0.0"/>
  <arg name="initial_pose_a" default="0.0"/>

  <!-- <node name="spawn_model" pkg="rosservice" type="rosservice" output="screen" 

  /> -->

  <!--  _____________ flatland viz  _____________  -->
  <group if="$(eval arg('visualization') == 'flatland')">
    <node name="flatland_viz" pkg="flatland_viz" type="flatland_viz" required="false" />
  </group>

  <!-- start_rviz -->
  <include file="$(find arena_bringup)/launch/utils/rviz.launch">
    <arg name="show_rviz" value="$(eval arg('visualization') == 'rviz')" />
  </include>
</launch>